src/main/java/com/spider/solomon/reporting/TrendAnalyzer.java

```diff
@@ -0,0 +1,51 @@
+package com.spider.solomon.reporting;
+
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+
+import org.apache.commons.math3.stat.regression.SimpleRegression;
+import org.springframework.stereotype.Component;
+
+import com.spider.solomon.order.CustomerOrder;
+import com.spider.solomon.order.CustomerOrderDao;
+
+import jakarta.annotation.Resource;
+
+/**
+ * Analyzes trends in sales data
+ */
+@Component
+public class TrendAnalyzer {
+	@Resource
+	CustomerOrderDao customerOrderDao;
+
+	/**
+	 * Returns whether the total sales data is trending up or down
+	 */
+	public String saleAmountTrend () {
+		SimpleRegression r = new SimpleRegression();
+
+		LocalDateTime tmp = customerOrderDao.getAllOrders().stream()
+			.map(CustomerOrder::getOrderDate)
+			.min(LocalDateTime::compareTo)
+			.orElse(LocalDateTime.MIN);
+
+		// Add in data for all orders to the SimpleRegression
+		customerOrderDao.getAllOrders().forEach(order -> {
```

Comment:
We're getting all the orders twice. Can we `getAllOrders` once and store the result, then we can iterate over it twice?

[Meta: Depending on the requirements, I might also question getting *all* orders. I imagine we'd probably just want to get some recent orders, maybe using a configurable start date]